### PR TITLE
JBDS-4293 Merge requirements-*.json files into one

### PR DESCRIPTION
**Pull request by dgolovin**

### CODE REVIEW

File: browser/services/data.js [hunk 1]
```diff
@@ -53,6 +53,27 @@ class InstallerDataService {
     }
   }
 
+  transformation(requirements) {
+    var obj = {};
+    for(var object in requirements) {
+      obj[object] = {};
+      for(var platform in requirements[object]) {
+        if(platform === 'platform') {
+          for(var operatingsystem in requirements[object][platform]) {
```
Comment (jrichter1) [inline browser/services/data.js]: The 'operatingsystem' loop has no use, remove it.

File: browser/services/data.js [hunk 2]
```diff
@@ -53,6 +53,27 @@ class InstallerDataService {
...
+            if(process.platform){
```
Comment (jrichter1) [inline browser/services/data.js]: Does this need to be a conditional? There doesn't seem to be an alternative to this (and if process.platform is not defined, than something else will fail anyway).

File: requirements.json [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    "vendor": "Igor Pavlov",
+    "platform": {
+      "win32": {
+        "bundle": "tools",
+        "version": "9.20",
+        "url": "https://downloads.sourceforge.net/project/sevenzip/7-Zip/9.20/7za920.zip?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fsevenzip%2Ffiles%2F7-Zip%2F9.20%2F",
+        "filename": "7zip.zip",
+        "sha256sum": "2a3afe19c180f8373fa02ff00254d5394fec0349f5804e0ad2f6067854ff28ac"
+      },
+      "linux": {
```
Comment (jrichter1) [inline requirements.json]: We don't need 7zip on linux, it's runs on windows only anyway.

File: requirements.json [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    "vendor": "Igor Pavlov",
+    "platform": {
+      "win32": {
+        "bundle": "tools",
+        "version": "9.20",
+        "url": "https://downloads.sourceforge.net/project/sevenzip/7-Zip/9.20/7z920_extra.7z?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fsevenzip%2Ffiles%2F7-Zip%2F9.20%2F",
+        "filename": "7zip-extra.7z",
+        "sha256sum": "d65a1d1a050e4fd7912c18468954f64db824a1e1b621235a153d010beae14757"
+      },
+      "linux": {
```
Comment (jrichter1) [inline requirements.json]: The same as base 7zip.

File: requirements.json (hunk 3 above)
Comment (sudhirverma) [inline requirements.json]: okay I will remove 7zip

File: browser/services/data.js (hunk 1 above)
Comment (sudhirverma) [inline browser/services/data.js]: yes, I will remove this loop :)

**Comment by dgolovin** [comment]
Sorry for confusion. I was working on this over weekend, but did not finished. Updated to what I wanted it to be.

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
test this

File: gulp-tasks/dist-darwin.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const download = require('./download.js');
-const reqs = require('../requirements-darwin.json');
+const reqs = require('../requirements.json');
```
Comment (jrichter1) [inline gulp-tasks/dist-darwin.js]: Any particular reason the reqs are loaded in a different fashion compared to windows?

Comment (dgolovin) [inline gulp-tasks/dist-darwin.js]: Good catch, that is definitely a bug. Thanks!

Comment (dgolovin) [inline gulp-tasks/dist-darwin.js]: Have not run macOS build yet, will do

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
@jrichter1 could you review? Thanks!

File: requirements.json [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        "filename": "linux-amd64.tar.gz",
+        "sha256sum": "76da07d4065eb387093af1316371a19c11982b4b9270eff14bca84db69e2bb32"
+      }
+    }
+  },
+  "cygwin": {
+    "name": "Cygwin",
+    "description": "A distribution of popular GNU and other Open Source tools running on Microsoft Windows",
+    "vendor": "Community Project",
+    "platform": {
```
Comment (jrichter1) [inline requirements.json]: I can't see any difference between the platforms here. Could it be done the way devstudio is?

Comment (dgolovin) [inline requirements.json]: You are correct, this is a code dupe.

**Comment by dgolovin** [comment]
test this

**Pull request merged**